src/Notifiers/Slack/SlackNotifier.php

```diff
@@ -0,0 +1,304 @@
+        $this->env = $env;
+        $this->httpClient = $httpClient;
+        $this->router = $router;
+        $this->conferenceRepository = $conferenceRepository;
+        $this->slackBlocksBuilder = $slackBlocksBuilder;
+    }
+
+    /** @param array<Conference> $newConferences */
+    public function sendDailyNotification(array $newConferences): void
+    {
+        $conferences = $this->buildConferencesBlocks($newConferences);
+        $endingCfps = $this->buildCfpsBlocks();
+
+        $blocks = [
+            'blocks' => [
+                ...$conferences,
+                ...$endingCfps,
+                $this->slackBlocksBuilder->buildDivider(),
+            ],
+        ];
+
+        $this->notify($blocks);
+    }
+
+    /** @param array<Submit> $submits */
+    public function sendNewTalkSubmittedNotification(array $submits, Talk $talk): void
+    {
+        $header = $this->slackBlocksBuilder->buildHeader('🗣  New submitted talk');
+        $title = $this->slackBlocksBuilder->buildSimpleSection(sprintf('*Title :* %s', $talk->getTitle()));
+        $intro = $this->slackBlocksBuilder->buildSimpleSection(sprintf('*Introduction :* %s', $talk->getIntro()));
+
+        $blocks = [
+            'blocks' => [
+                $header,
+                $this->slackBlocksBuilder->buildDivider(),
+                $title,
+                $intro,
+            ],
+        ];
+
+        if (0 < \count($submits)) {
+            $submittedText = '*Submitted for :*';
+            $index = 1;
+
+            foreach ($submits as $submit) {
+                $conference = sprintf('<%s|%s>', $submit->getConference()->getSiteUrl(), $submit->getConference()->getName());
+                $status = Submit::STATUS_EMOJIS[$submit->getStatus()];
+                $author = $submit->reduceSpeakersNames();
+
+                if (1 === $index) {
+                    $submittedText .= sprintf(' %s (%s) by %s', $conference, $status, $author);
+                } else {
+                    $submittedText .= sprintf(', %s (%s) by %s', $conference, $status, $author);
+                }
+
+                ++$index;
+            }
+
+            $blocks['blocks'][] = $this->slackBlocksBuilder->buildSimpleSection($submittedText);
+
+            $talkRoute = $this->router->generate('easyadmin', [
+                'id' => $talk->getId(),
+                'entity' => 'Talk',
+                'action' => 'show',
+            ], UrlGeneratorInterface::ABSOLUTE_URL);
+
+            $blocks['blocks'][] = $this->slackBlocksBuilder->buildDivider();
+            $blocks['blocks'][] = $this->slackBlocksBuilder->buildContext(sprintf('<%s|See the talk on Starfleet>', $talkRoute));
+        }
+
+        $this->notify($blocks);
+    }
+
+    public function sendSubmitStatusChangedNotification(Submit $submit): void
+    {
+        if (Submit::STATUS_ACCEPTED === $submit->getStatus()) {
+            $header = $this->slackBlocksBuilder->buildHeader('🎉  Talk accepted');
+        } else {
```

Comment:
Be careful here, a Talk could be updated to go back into "pending" status, it has already happened in the past

Reply:
It's not visible here but this case is handled in the SlackNotifierEventListener : https://github.com/jolicode/starfleet/blob/master/src/EventListener/SlackNotifierEventListener.php#L78